rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java

```diff
@@ -39,4 +48,24 @@ protected String getBarcodeValsAsCSV(Intent data) {
     protected String[] splitCSV(String barcodeCSV) {
         return barcodeCSV.split(",");
     }
-}
+
+    @Override
+    protected void moveToNextStep(Intent data, Date expDate) {
+        final int sensorTriggerIndex = 4;
+        Barcode barcode = data.getParcelableExtra(BARCODE_KEY);
+        if (Boolean.parseBoolean(barcode.displayValue.split(",")[sensorTriggerIndex])) {
+            JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
+            String expiredPageAddr = stepStateConfig.getStepStateObj().optString(Constants.Step.PRODUCT_EXPIRED_PAGE, "step1");
+            try {
+                jsonApi.writeValue(expiredPageAddr, "lbl_sample_expired", "",  "", "", "", false);
+                jsonApi.writeValue(expiredPageAddr, "lbl_collect_new_sample", widgetArgs.getContext().getString(R.string.rdt_high_heat_exposure),  "", "", "", false);
```

Comment:
This is not required. The text should be modified within the form step2.

Reply:
you mean to add another field with a message in step2 and add the relevance check with step6 temp_sensor.